test/suite_oc-desktop/shared/steps/steps.py

```diff
@@ -0,0 +1,343 @@
+        clickButton(waitForObject(names.scrollArea_permissionsEdit_QCheckBox))
+    if ('share' in permissionsList and shareChecked == False) or ('share' not in permissionsList and shareChecked == True):
+        clickButton(waitForObject(names.scrollArea_permissionShare_QCheckBox))
+    
+    clickButton(waitForObject(names.sharingDialog_Close_QPushButton))
+
+@Then('user "|any|" should be listed in the collaborators list for file "|any|" with permissions "|any|" on the client-UI')
+def step(context, receiver, resource, permissions):
+    resource = substituteInLineCodes(context, resource)
+    socketConnect = syncstate.SocketConnect()
+    socketConnect.sendCommand("SHARE:" + resource + "\n")
+    permissionsList = permissions.split(",")
+    
+    test.compare(str(waitForObjectExists(names.scrollArea_sharedWith_QLabel).text), receiver)
+    test.compare(waitForObjectExists(names.scrollArea_permissionsEdit_QCheckBox).checked, ('edit' in permissionsList))
+    test.compare(waitForObjectExists(names.scrollArea_permissionShare_QCheckBox).checked, ('share' in permissionsList))
+
+@When("the user waits for the files to sync")
+def step(context):
+    waitFor(lambda: isFolderSynced(context.userData['clientSyncPath']), context.userData['clientSyncTimeout'] * 1000)
+
+@When("the user waits for file '|any|' to get synced")
+def step(context, fileName):
+    waitForFileToBeSynced(context, fileName)
+
+@Given("the user has waited for file '|any|' to get synced")
+def step(context, fileName):
+    waitForFileToBeSynced(context, fileName)
+
+@When("the user creates a file '|any|' with following content on the file system")
+def step(context, filename):
+    fileContent = "\n".join(context.multiLineText)
+    f = open(context.userData['clientSyncPath'] + filename, "w")
+    f.write(fileContent)
+    f.close()
+
+@Given("user '|any|' has uploaded file with content '|any|' to '|any|'")
+def step(context, username, fileContent, filename):
+    executeStepThroughMiddleware(
+        context,
+        "Given user \"" + username + "\" has uploaded file with content \"" +
+        fileContent + "\" to \"" + filename + "\"",
+        'Failed to upload file'
+    )
+
+@Given("user '|any|' has been created with default attributes")
+def step(context, username):
+    executeStepThroughMiddleware(
+        context,
+        "Given user \"" + username + "\" has been created with default attributes",
+        "Failed to create user"
+    )
+
+@Then("the file '|any|' should exist on the server for user '|any|' with following content")
+def step(context, path, username):
+    fileContent = "\n".join(context.multiLineText)
+    executeStepThroughMiddleware(
+        context,
+        "Then as \"" + username + "\" the file \"" + path + "\" should have the content \"" + fileContent + "\"",
+        "Failed getting file from server or comparing the content"
+    )
```

Comment:
Since the middleware server is responsible for parsing the step and run the appropriate step definition, I would prefer if there was a common step definition here to execute a step on middleware

Reply:
addressed in e1177ee2839256afebec6a85ff2ea492b7fef475